model/riscv_insts_vext_fp.sail

```diff
@@ -0,0 +1,1311 @@
+
+function clause execute(FWVTYPE(funct6, vm, vs2, vs1, vd)) = {
+  let rm_3b    = fcsr.FRM();
+  let SEW      = get_sew();
+  let LMUL_pow = get_lmul_pow();
+  let num_elem = get_num_elem(LMUL_pow, SEW);
+  let SEW_widen      = SEW * 2;
+  let LMUL_pow_widen = LMUL_pow + 1;
+
+  if  not(valid_rd_mask(vd, vm)) | not(valid_reg_overlap(vs1, vd, LMUL_pow, LMUL_pow_widen)) | 
+      not(valid_eew_emul(SEW_widen, LMUL_pow_widen)) | not(valid_fp_op(SEW, rm_3b)) 
+  then { handle_illegal(); return RETIRE_FAIL };
+  assert(SEW >= 16 & SEW_widen <= 64);
+
+  let 'n = num_elem;
+  let 'm = SEW;
+  let 'o = SEW_widen;
+
+  let vm_val  : vector('n, dec, bool)     = read_vmask(num_elem, vm, vreg_name("v0"));
+  let vd_val  : vector('n, dec, bits('o)) = read_vreg(num_elem, SEW_widen, LMUL_pow_widen, vd);
+  let vs1_val : vector('n, dec, bits('m)) = read_vreg(num_elem, SEW, LMUL_pow, vs1);
+  let vs2_val : vector('n, dec, bits('o)) = read_vreg(num_elem, SEW_widen, LMUL_pow_widen, vs2);
+  result      : vector('n, dec, bits('o)) = undefined;
+  mask        : vector('n, dec, bool)     = undefined;
+
+  (result, mask) = init_masked_result(num_elem, SEW_widen, LMUL_pow_widen, vd_val, vm_val);
+
+  foreach (i from 0 to (num_elem - 1)) {
+    if mask[i] then {
+      result[i] = match funct6 {
+        FWV_VADD     => fp_add(rm_3b, vs2_val[i], fp_widen(vs1_val[i])),
+        FWV_VSUB     => fp_sub(rm_3b, vs2_val[i], fp_widen(vs1_val[i]))
+      }
+    }
+  };
+
+  write_vreg(num_elem, SEW_widen, LMUL_pow_widen, vd, result);
+  vstart = EXTZ(0b0);
+  RETIRE_SUCCESS
+}
+
+mapping fwvtype_mnemonic : fwvfunct6 <-> string = {
+  FWV_VADD       <-> "vfwadd.wv",
+  FWV_VSUB       <-> "vfwsub.wv"
+}
+
+mapping clause assembly = FWVTYPE(funct6, vm, vs2, vs1, vd)
+  <-> fwvtype_mnemonic(funct6) ^ spc() ^ vreg_name(vd) ^ sep() ^ vreg_name(vs2) ^ sep() ^ vreg_name(vs1) ^ maybe_vmask(vm)
+
+/* ****************************** OPFVV (VFUNARY0) ******************************* */
+union clause ast = VFUNARY0 : (bits(1), regidx, vfunary0, regidx)
+
+mapping encdec_vfunary0_vs1 : vfunary0 <-> bits(5) = {
+  FV_CVT_XU_F       <-> 0b00000,
+  FV_CVT_X_F        <-> 0b00001,
+  FV_CVT_F_XU       <-> 0b00010,
+  FV_CVT_F_X        <-> 0b00011,
+  FV_CVT_RTZ_XU_F   <-> 0b00110,
+  FV_CVT_RTZ_X_F    <-> 0b00111
+}
+
+mapping clause encdec = VFUNARY0(vm, vs2, vfunary0, vd) if haveRVV()
+  <-> 0b010010 @ vm @ vs2 @ encdec_vfunary0_vs1(vfunary0) @ 0b001 @ vd @ 0b1010111 if haveRVV()
+
+function clause execute(VFUNARY0(vm, vs2, vfunary0, vd)) = {
+  let rm_3b    = fcsr.FRM();
+  let SEW      = get_sew();
+  let LMUL_pow = get_lmul_pow();
+  let num_elem = get_num_elem(LMUL_pow, SEW);
+
+  if not(valid_rd_mask(vd, vm)) | not(valid_fp_op(SEW, rm_3b)) then { handle_illegal(); return RETIRE_FAIL };
+
+  /* Special case: 16-bit integers are not supported in single-width type-convert instructions */
+  if SEW == 16 then { handle_illegal(); return RETIRE_FAIL };
```

Comment:
I don't understand the reason for this restriction. Conversion for SEW16 are supported in the VSpec.

Reply:
Conversions involving 16-bit integers are not supported in Softfloat. I checked the Spike repo and found that they add this support by themselves. So I will add this part by using their extended softfloat functions.

Reply:
This support is added now